packages/nuxt/src/core/app.ts

```diff
@@ -123,6 +123,7 @@ export async function resolveApp (nuxt: Nuxt, app: NuxtApp) {
   for (const config of layerConfigs) {
     const layoutDir = (config.rootDir === nuxt.options.rootDir ? nuxt.options : config).dir?.layouts || 'layouts'
     const layoutFiles = await resolveFiles(config.srcDir, `${layoutDir}/**/*{${nuxt.options.extensions.join(',')}}`)
+    app._hasLayouts = layoutFiles.length > 0
```

Comment:
I think you can rely on `app.layouts` later on instead - we do not support `~/layouts/index.vue` as a layout at all.